src/science/canopy/cbl_LAI_canopy_height.F90

```diff
@@ -1,26 +1,66 @@
 module cbl_LAI_canopy_height_mod
+  !* This MODULE contains the SUBROUTINE [[limit_HGT_LAI]].  This routine
+  ! places limits on the values of the leaf area and canopy height to ensure
+  ! realism.
 
 contains
 
 subroutine limit_HGT_LAI( HGT_pft_temp, LAI_pft_cbl, HGT_pft_cbl, mp, land_pts, ntiles, &
                           tile_pts, tile_index, tile_frac,L_tile_pts, &
                           LAI_pft, HGT_pft, CLAI_thresh )
-
+  !*## Purpose
+  !
+  ! This SUBROUTINE checks that input values of leaf area and canopy height
+  ! lie between set values (defined by PFT) and overwrites if necessary.
+  ! These limits are needed to ensure that the canopy roughness properties of
+  ! the land point are well behaved (see [[ruff_resist]]).
+  !
+  ! The mapping from grid cells (land_pts,ntiles) to the (mp) structure
+  ! is also undertaken.
+  !
+  ! **This SUBROUTINE is active run when CABLE is run within a coupled model
+  ! (i.e. ACCESS)** 
+  !
+  !## Method
+  !
+  ! The cell-level input (land_pts,ntile) variables for leaf area `LAI_pft` and
+  ! canopy height `HGT_pft` are checked for
+  !
+  ! - whether the land_pt has a non-zero fraction of that tile (if zero
+  !   fraction then LAI and canopy height are set to zero.
+  ! - whether the LAI lies above a minimum value `CLAI_thresh`
+  ! - whether the canopy height lies a minimum value (which are PFT dependent)
+  ! - whether non-vegetated tiles have a non-zero canopy height.
+  !
+  ! Outputs are `LAI_pft_cbl` and `HGT_pft_cbl`
+  !
+  ! **WARNINGS**
+  !
+  ! - INTENT statements need to be added to the argument lists
+  ! - hardwired indexing is use throughout.  This SUBROUTINE assumes that
+  !     1. non-vegetated tiles take indexes 14 and onwards
+  !     2. tall vegetation occupies tile indexes 1-4
+  !     3. other vegetation (shrubs/grasses/crops/wetlands) occupy tile indexes
+  !     5-13
+  ! - the limits on canopy height are hardwired
+  !     1. canopy height for tall vegetation is limited to be greater than 1m
+  !     2. canopy height for other vegetation is limited to be greater than 0.1m
+  
 implicit none
-real :: Clai_thresh                 !The minimum LAI below which a "cell" is considred NOT vegetated
-integer :: land_pts, ntiles
-integer :: tile_pts(ntiles) 
-integer:: tile_index(land_pts,ntiles) 
-real:: tile_frac(land_pts,ntiles) 
-logical :: L_tile_pts(land_pts,ntiles) 
-integer :: mp
-real :: LAI_pft(land_pts, ntiles)
-real :: HGT_pft(land_pts, ntiles)
+real :: Clai_thresh                 !! The minimum LAI below which a tile is considered to be NOT vegetated (m\(^2\)m\(^{-2}\))
+integer :: land_pts, ntiles         !! number of land points, max number of tiles that can be carried in any land point (-)
+integer :: tile_pts(ntiles)         !! number of land_pts that a particular tile occupies (-) 
```

Comment:
1 tile can cover several land points??? I don't think I understand here.

Reply:
okay - this is a bit confused.  I THINK that 

- each land_pt has up to max number of tiles as subunits (ntiles)
- but ntiles < number of tile types considered (=17 currently)

tile_pts(ntiles) is then number of land_pts which include a tile with a particular value of the tile type and tile_index is the mapping between (land_pts,ntile) and the tile type.

I will attempt to revise that comment accordingly